rfcs/0012-compaction-state-persistence.md

```diff
@@ -0,0 +1,800 @@
+      ( The current active compactor Job would have updated the .compactor file)
+
+4. If compactor_epoch in in-memory manifest(00005.manifest) >= `compactor_epoch`, the compactor is fenced.
+
+5. Try writing the above `compaction_epoch` to the .manifest file (00006.manifest)
+
+    File version check (in-memory and remote object store), If 00006.manifest exists, 
+
+      - If latest .manifest compactor_epoch > current compactor epoch, die (fenced)
+
+      - If latest .manifest compactor_epoch == current compactor epoch, panic
+
+      - If latest .manifest compactor_epoch < current compactor epoch, increment the .manifest file ID by 1 and retry. This process would continue until successful compactor write.
+      ( The current active compactor Job would have updated the .manifest file)
+
+At this point, the compactor has been successfully initialised. Any updates to write a new .compactor(00006.compactor) or .manifest file(00006.manifest) by stale compactors would fence them.
+
+#### On compaction initiation...
+
+(Manifest is polled periodically to get the list of L0 SSTs created. The scheduler would create a list of new compactions for these L0 SSTs as well)
+
+1. Compactor Fetches the latest .manifest file during the manifest poll.(00006.manifest)
+
+2. Compactor writes to the next .compactor file the list of scheduled compactions with the empty JobAttempts (00007.compactor in our example).
+
+    If the file(00007.compactor) exists, 
+
+      - If latest .compactor compactor_epoch > current compactor epoch, die (fenced)
+
+      - If latest .compactor compactor_epoch == current compactor epoch, reconcile the compactor state and go to step (2)
+
+      - If latest .compactor compactor_epoch < current compactor epoch, panic
+        (compactor_epoch going backwards)
+
+#### On compaction job progress...
+
+1. Compactor writes to the next .compactor file the compactionState(persist when an SST is added to SR) with the latest progress (00008.compactor in our example).
+
+    If the file(00008.compactor) exists, 
+
+      - If latest .compactor compactor_epoch > current compactor epoch, die (fenced)
+
+      - If latest .compactor compactor_epoch == current compactor epoch, reconcile the compactor state and go to step (2)
+
+      - If latest .compactor compactor_epoch < current compactor epoch, panic
+        (compactor_epoch going backwards)
+
+#### On compaction job complete...
+
+1. Write the current compactor state (including the completed compaction job) to the next sequential .compactor file(00009.compactor) (steps (1) and (2) in the "progress" section, above).
+
+2. Update in-memory .manifest state (fetched in compaction initiation phase) with the compaction state to reflect the latest SRs/SSTs that were created (and remove old SRs/SSTs).
+
+3. Write the in-memory .manifest state to the next sequential .manifest file. If the file(00007.manifest) exists, it could be due to two possibilities:
```

Comment:
```suggestion
3. Write the in-memory .manifest state to the next sequential .manifest file. If the file (00007.manifest) exists, it could be due to three possibilities:
```